src/engine/engine_bf.c

```diff
@@ -96,49 +106,40 @@ static int _ocf_backfill_do(struct ocf_request *req)
 		return 0;
 	}
 
+	seek = req->addr % cache_line_size;
+	last_chunk_size = (req->addr + req->bytes) % cache_line_size;
+	skip = (cache_line_size - last_chunk_size) % cache_line_size;
+
 	ocf_req_forward_cache_get(req);
 	for (i = 0; i < req->core_line_count; i++) {
-		if (addr_next) {
-			addr = addr_next;
-		} else {
-			addr  = req->map[i].coll_idx;
-			addr *= ocf_line_size(cache);
-			addr += cache->device->metadata_offset;
-		}
-		bytes = ocf_line_size(cache);
-
-		if (i == 0) {
-			seek = req->addr % ocf_line_size(cache);
-			addr += seek;
-			bytes -= seek;
-		}
-
 		if (req->map[i].status == LOOKUP_HIT) {
-			/* This is the 1st cache line in the interval,
-			 * and it's a hit. Don't write it to the cache.
-			 */
-			addr_next = 0;
-			total_bytes += bytes;
+			/* This is the 1st cache line in the interval, and it's
+			   a hit. Don't write it to the cache */
+			total_bytes += cache_line_size;
+			total_bytes -= seek;
+			/* Seek should be taken into account for the first chunk
+			   only */
+			seek = 0;
```

Comment:
I'd prefer `unlikely()` on `if (i == 0)`. Now we are zeroing this `seek` here and there many times for completely no reason.